pkg/api/mlflow/service/artifact/storage/s3.go

```diff
@@ -121,3 +125,27 @@ func (s S3) List(artifactURI, path string) ([]ArtifactObject, error) {
 
 	return artifactList, nil
 }
+
+// Get returns file content at the storage location.
+func (s S3) Get(artifactURI, itemPath string) (io.ReadCloser, error) {
+	bucketName, prefix, err := ExtractS3BucketAndPrefix(artifactURI)
+	if err != nil {
+		return nil, eris.Wrap(err, "error extracting bucket and prefix from provided uri")
+	}
+
+	input := &s3.GetObjectInput{
+		Bucket: aws.String(bucketName),
+		Key:    aws.String(filepath.Join(prefix, itemPath)),
+	}
+
+	resp, err := s.client.GetObject(context.TODO(), input)
+	if err != nil {
+		var s3NoSuchKey *types.NoSuchKey
+		if errors.As(err, &s3NoSuchKey) {
```

Comment:
maybe `is` will work better/shorter? you just need to make sure that err has type of `type.NoSuchKey` . you don't need then to use this error, so maybe `is`?

Reply:
following example from the aws sdk docs, `is` does not work

Reply:
I have tried as well but haven't found a way to get this to work with a regular `errors.Is`. It seems to be because the error is a pointer _and_ does not implement `Is()`.